### ISSUE TITLE
Create a 'getting started' guide for developers

### ISSUE THREAD

**Issue by samccann** [ansible/ansible#71601]
##### SUMMARY

This came up as a possible ansibleFest topic, so thought it would be worth having a specific getting started guide for developers so they can quickly learn how to:

- get ansible from source and set up your basic development environment
- Open a PR for a bugfix or new option to a module
- how to find the module source code in the collections world
- PR requirements like changelogs, unit tests, etc
- keeping your PR current for  newer versions of the upstream repo 

(git remote add ; git fetch ; git merge --ff-only to devel  ; git checkout feature-branch ; git rebase devel )

##### ISSUE TYPE
- Documentation Report

##### COMPONENT NAME

docs.ansible.com
##### ANSIBLE VERSION

```paste below

```

##### CONFIGURATION

```paste below

```

**Comment by samccann** [comment]
https://github.com/greg-hellings/ansible_collection_demo - has some info that might be helpful for starting a new collection.

**Comment by s-hertel** [comment]
@samccann Can you move this to the new repo?

**Comment by samccann** [comment]
yes I plan on going through all the docs issues after June 10th to move them all over. That'  the proposed 'official' date when the split happens. I can try to do it sooner, but I'm out from Friday until that date so didn't want to rush it all.

**Comment by s-hertel** [comment]
Thanks, this was just in our backlog queue.
